EIPS/eip-1052.md

```diff
@@ -18,16 +18,57 @@ Many contracts need to perform checks on a contract's bytecode, but do not neces
 Contracts can presently do this using the `EXTCODECOPY` opcode, but this is expensive, especially for large contracts, in cases where only the hash is required. As a result, we propose a new opcode, `EXTCODEHASH`, which returns the keccak256 hash of a contract's bytecode.
 
 ## Specification
-A new opcode, `EXTCODEHASH`, is introduced, with number 0x3D. `EXTCODEHASH` takes one argument from the stack, and pushes the keccak256 hash of the code at that address to the stack.
+
+A new opcode, `EXTCODEHASH`, is introduced, with number 0x3D. The `EXTCODEHASH` 
+takes one argument from the stack, zeros the first 96 bits 
+and pushes to the stack the keccak256 hash of the code of the account 
+at the address being the remaining 160 bits. 
+
+In case the account does not exist `0` is pushed to the stack.
+
+In case the account does not have code the keccak256 hash of empty data
+(i.e. `c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`)
+is pushed to the stack.
+
+The gas cost of the `EXTCODEHASH` is 400.
+
 
 ## Rationale
-As described in the motivation section, this opcode is widely useful, and saves on wasted gas in many cases.
+
+As described in the motivation section, this opcode is widely useful, and saves 
+on wasted gas in many cases.
+
+The gas cost is the same as the gas cost for the `BALANCE` opcode because the 
+execution of the `EXTCODEHASH` requires the same account lookup as in `BALANCE`.
+
+Only the 20 last bytes of the argument are significant (the first 12 bytes are 
+ignored) similarly to the semantics of the `BALANCE`, `EXTCODESIZE` and 
+`EXTCODECOPY`.
+
+The `EXTCODEHASH` distincts accounts without code and non-existing accounts.
+This is consistent with the way accounts are represented in the state trie.
+This also allows smart contracts to check whenever an account exists.
+
 
 ## Backwards Compatibility
+
 There are no backwards compatibility concerns.
 
+
 ## Test Cases
-TBD
+
+1. The `EXTCODEHASH` of the account without code is `c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`
+   what is the keccack256 hash of empty data.
+2. The `EXTCODEHASH` of non-existent account is `0`.
+3. The `EXTCODEHASH` of an precompiled contract is either `c5d246...` or `0`.
+4. If `EXTCODEHASH` of `A` is `X`, then `EXTCODEHASH` of `A + 2**160` is `X`.
+5. The `EXTCODEHASH` of an account that selfdestructed in the current transaction.
```

Comment:
What it supposed to happen here? The EIP says these are test cases but then never provides the expected results! Currently trinity and geth handle this situation differently, who's to say who is right if not the EIP?

@Arachnid, once I figure out what should happen should I open a PR against this EIP? Or, since it's final, should I open a new clarifying EIP?